base/io.jl

```diff
@@ -60,9 +60,36 @@ function isopen end
 Close an I/O stream. Performs a [`flush`](@ref) first.
 """
 function close end
+
+"""
+    shutdown(stream)
+
+Shutdown an I/O stream. Performs a [`flush`](@ref) first. Notify the other end that no
+more data can ever be written to the underlying file. This is not supported by all IO
+types.
+"""
```

Comment:
Suggested wording tweak:
"""
Shutdown _**the write half of a full-duplex**_ I/O stream. Performs a flush first. Notify the other end that no more data _**will be written to the stream**_. This is not supported by all IO types.
"""
Perhaps include a pseudo-code example?
```
write(io, request)
shutdown(io)
result = read(io)
```